dlib/CMakeLists.txt

```diff
@@ -972,6 +972,11 @@ if (MSVC)
    set_target_properties(dlib PROPERTIES RELWITHDEBINFO_POSTFIX "${VERSION}_relwithdebinfo_${numbits}bit_msvc${MSVC_VERSION}")
 endif()
 
+if (NOT MSVC)
+   # Treat warnings as errors.
+   target_compile_options(dlib PRIVATE -Werror)
```

Comment:
This makes me paranoid.  How about putting this in `test/CMakeLists.txt` instead?  Otherwise what's invariably going to happen is microsoft will release a new version of visual studio with some crazy new nonsense warning and people will then start posting "dlib doesn't compile on visual studio 2024!"

Reply:
You did notice the `if (NOT MSVC)` part, right? :)

But yes, this PR [already contains](https://github.com/davisking/dlib/pull/2490/files#diff-6845460d9ecba59453be723566579dffe7bb9bc9fd4fae82bba498a45642e91e) this in `test/CMakeLists.txt`. So I can remove this from here, I think. I'll just first make sure that the [original warning](https://github.com/davisking/dlib/pull/2483) that spurred the need for this PR will still be caught in that case.

Reply:
Yes, we're still getting those warnings (even if we don't say `-Werror` here):
https://github.com/reunanen/dlib/runs/5077967832

Reply:
Heh, right, lost the thread here amongst other emails.  Still though, same problem for any other compiler.  We shouldn't make dlib not compile on client software builds just because some new warning appears for whatever compiler they might be using.

Reply:
Yes, understand completely. Warnings are now enabled for tests only.